Lib/py_compile.py

```diff
@@ -137,6 +137,10 @@ def compile(file, cfile=None, dfile=None, doraise=False, optimize=-1):
     except FileExistsError:
         pass
     source_stats = loader.path_stats(file)
+    sde = os.environ.get('SOURCE_DATE_EPOCH')
+    if sde and source_stats['mtime'] > int(sde):
```

Comment:
The traceback if `int` does not tell the end user to define the `SOURCE_DATE_EPOCH' environment variable with a valid integer value.